app/utils/lava_log_parser.py

```diff
@@ -122,27 +124,33 @@ def run(log, boot, txt, html):
 
     start_ts = None
     log_buffer = []
+    lineno = 0
 
     for line in log:
+        lineno = lineno + 1
         dt, level, msg = (line.get(k) for k in ["dt", "lvl", "msg"])
         raw_ts = DT_RE.match(dt).groups()[1]
         timestamp = "<span class=\"timestamp\">{}  </span>".format(raw_ts)
+        sl = "<p id=%d>" % lineno
+        el = "</p\n>"
 
         if isinstance(msg, list):
             msg = ' '.join(msg)
 
         fmt = formats.get(level)
         if fmt:
-            log_buffer.append(timestamp + fmt.format(cgi.escape(msg)))
+            log_buffer.append(
```

Comment:
I think that with that number of elements, using `join()` is a bit more efficient:
```python
''.join([sl, timestamp, fmt.format(cgi.escape(msg)), el])
```